### PR TITLE
pkg/u8g2: refactoring and fixes

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description
This is an updated version of #10693, with the comments fixed and some improvements to code style, naming of methods and documentation. Still, most work is done by @pvyleta.

I also updated to the latest version of U8g2.

This version will slightly change the RIOT-OS interface for U8g2, but it will save bytes. The test has been updated to reflect this change.

As a bonus, this PR doesn't need any patches anymore, so I think it will be easier to maintain.

### Testing procedure
The test that outputs to a virtual screen (UTF8) can be tested using `make -C tests/pkg_u8g2 all test`.

To test the SPI/I2C, you need to connect a display. I'm using the cheap SSD1306 OLED displays and tested a I2C version and a SPI version using the STK3600.

### Issues/PRs references
#10693

### CODE REVIEW

**Comment by basilfx** [comment]
@aabadie I did not [change the way](https://github.com/RIOT-OS/RIOT/pull/13036#issuecomment-571756127) the package is build, because I don't know how it works with optional modules.

**Comment by basilfx** [comment]
Here is some pictures of the two displays I tested:

SSD1306 SPI:
EBEE781A-366F-4412-88E9-363E4BED206B_1_105_c

SSD1306 I2C:
778DAFB8-29E7-43E5-9F54-FD6B0DB6DE7B_1_105_c

File: pkg/u8g2/Makefile.dep [hunk 1]
```diff
@@ -0,0 +1,5 @@
+USEMODULE += xtimer
+
+FEATURES_REQUIRED += periph_gpio
+
+CFLAGS += -DU8X8_WITH_USER_PTR
```
Comment (aabadie) [inline pkg/u8g2/Makefile.dep]: This should go in a separate `Makefile.include` file.

**Comment by basilfx** [comment]
> You could add 2 pseudomodule, u8g2-spi and u8g2-i2c, that would depend on the u8g2 package and periph_spi/periph_i2c features.

You mean the Makefile.dep in the root of RIOT-OS? Now everything is self-contained, which isn't the case with pseudomodules.

File: pkg/u8g2/Makefile.dep (hunk 1 above)
Comment (basilfx) [reply pkg/u8g2/Makefile.dep]: Moved.

**Comment by basilfx** [comment]
@aabadie I have made an attempt in moving it in an other structure (locally), but I am not yet convinced. 

I'd rather have this PR being merged first, and open a different PR for setting it up. I might need your help there ;-)

**Comment by aabadie** [comment]
> I have made an attempt in moving it in an other structure (locally), but I am not yet convinced.

That's fine. It was just a suggestion.

**Comment by basilfx** [comment]
@aabadie Squashed and rebased.

**Pull request merged**